mono/metadata/threadpool-worker-default.c

```diff
@@ -356,6 +356,25 @@ mono_threadpool_worker_request (void)
 	mono_refcount_dec (&worker);
 }
 
+#if ENABLE_NETCORE
+gint64 mono_threadpool_worker_get_completed_threads_count (void)
+{
+	return worker.heuristic_completions;
+}
+
+gint64 mono_threadpool_worker_get_pending_threads_count (void)
+{
+	ThreadPoolWorkerCounter const counter = COUNTER_READ ();
+	return counter._.starting;
```

Comment:
I don't think this is right:
1. this is called for `GetPendingUnmanagedWorkItemCount` not for pending threads.
2. The CoreCLR impl is counting tasks I think, not worker threads that are just starting up: https://github.com/dotnet/coreclr/blob/9bd2787a9dd2aa4d2b7d4f72afebc3dbe896e896/src/vm/comthreadpool.cpp#L208-L213

Reply:
@lambdageek 
```csharp
Console.WriteLine($"1. tc:{ThreadPool.ThreadCount},comp:{ThreadPool.CompletedWorkItemCount},pend:{ThreadPool.PendingWorkItemCount}");
ThreadPool.QueueUserWorkItem(_ => {  while (true) ; });
Console.WriteLine($"2. tc:{ThreadPool.ThreadCount},comp:{ThreadPool.CompletedWorkItemCount},pend:{ThreadPool.PendingWorkItemCount}");
```
With my changes it prints the same what coreclr does

Reply:
However, I am still checking...

Reply:
CoreRT doesn't implement it: https://github.com/dotnet/corert/blob/c880f866597fc5b6f9348c60ee21b63ff2a893de/src/System.Private.CoreLib/src/System/Threading/ThreadPool.CoreRT.cs#L58

Reply:
I think we should also return 0.

The only instance I found so far in CoreCLR where they queue an unmanaged work item is in their tiered JIT: https://github.com/dotnet/coreclr/blob/060c8a228fbe397b4207eaed2aedff7bbbf32586/src/vm/tieredcompilation.cpp#L550

(That ends up calling QueueUnmanagedWorkRequest which increments `m_NumRequests` which is the counter that's returned by `GetPendingUnmanagedWorkItemCount`)

Reply:
I agree